search.go

```diff
@@ -55,7 +55,9 @@ func searchHandlerWithProxyMode(logger *zap.Logger, indexer Indexer, proxyMode *
 				return
 			}
 			packages = packages.Join(proxiedPackages)
-			packages = latestPackagesVersion(packages)
+			if !opts.Filter.AllVersions {
+				packages = latestPackagesVersion(packages)
```

Comment:
Do we have tests covered by this? I would be interested on seeing that we don't duplicate packages, but `packages.Join` should take care of this.